wgpu-types/src/lib.rs

```diff
@@ -881,56 +1191,90 @@ pub enum SwapChainStatus {
     OutOfMemory,
 }
 
+/// Operation to perform to the output attachment at the start of a renderpass.
+///
+/// This must be clear if it is the first renderpass rendering to a swap chain image.
 #[repr(C)]
 #[derive(Copy, Clone, Debug, Hash, Eq, PartialEq)]
 #[cfg_attr(feature = "trace", derive(Serialize))]
 #[cfg_attr(feature = "replay", derive(Deserialize))]
 #[cfg_attr(feature = "peek-poke", derive(PeekPoke))]
 pub enum LoadOp {
+    /// Clear the output attachment with the clear color.
     Clear = 0,
+    /// Do not clear output attachment. When rendering to a swap chain image, you can only use this
+    /// when a previous renderpass has cleared the swap chain image.
     Load = 1,
 }
 
+/// Operation to perform to the output attachment at the end of a renderpass.
 #[repr(C)]
 #[derive(Copy, Clone, Debug, Hash, Eq, PartialEq)]
 #[cfg_attr(feature = "trace", derive(Serialize))]
 #[cfg_attr(feature = "replay", derive(Deserialize))]
 #[cfg_attr(feature = "peek-poke", derive(PeekPoke))]
 pub enum StoreOp {
+    /// Clear the render target
     Clear = 0,
+    /// Store the result of the renderpass
     Store = 1,
 }
 
+/// Describes a color attachment to a [`RenderPass`].
 #[repr(C)]
 #[derive(Clone, Debug)]
 #[cfg_attr(feature = "trace", derive(Serialize))]
 #[cfg_attr(feature = "replay", derive(Deserialize))]
 #[cfg_attr(feature = "peek-poke", derive(PeekPoke))]
 pub struct RenderPassColorAttachmentDescriptorBase<T> {
+    /// Texture attachment to render to. Must contain [`TextureUsage::OUTPUT_ATTACHMENT`].
     pub attachment: T,
+    /// MSAA resolve target. Must contain [`TextureUsage::OUTPUT_ATTACHMENT`]. Must be `None` if
+    /// attachment has 1 sample (does not have MSAA).
     pub resolve_target: Option<T>,
```

Comment:
As I learned in the last hours, this paramter seems not to be mandatory even if you use MSAA. Maybe add a paragraph about what effect it has and how its supposed to be used?